### PR TITLE
Proper BinarySerializer & StructuredSerializer implementations for Event

### LINKED ISSUE #41: BinarySerializer/StructuredSerializer should be implemented for EventBuilder, not Event
Reimplement `BinarySerializer` and `StructuredSerializer` for `EventBuilder`, removing the implementation for `Event`

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Fix #41

- Add two `BinarySerializer` and `StructuredSerializer` for `Event` based on `EventBuilder` trait, in order to avoid creating event with defaults
- Now `Event` doesn't implement directly `BinarySerializer` and `StructuredSerializer`.
- Clean a bit the code

### CODE REVIEW

File: src/event/message.rs [hunk 1]
```diff
@@ -51,23 +51,29 @@ pub(crate) trait AttributesSerializer {
...
 
-enum EventBinarySerializer {
+#[derive(Debug)]
+pub(crate) enum EventBinarySerializer {
     V10(EventBuilderV10),
     V03(EventBuilderV03)
 }
 
+impl EventBinarySerializer {
+    pub(crate) fn new() -> Self {
```
Comment (pranav-bhatt) [inline src/event/message.rs:70]: Just confirming, is it fine to ignore v03 here?

File: src/event/data.rs [hunk 2]
```diff
@@ -46,7 +46,7 @@ impl Data {
 }
 
 pub(crate) fn is_json_content_type(ct: &str) -> bool {
-    ct == "application/json" || ct == "text/json" || ct.ends_with("+json")
```
Comment (pranav-bhatt) [inline src/event/data.rs:49]: Why change it over here?

File: src/event/message.rs (hunk 1 above)
Comment (slinkydeveloper) [reply src/event/message.rs:70]: Yes because then with `set_spec_version` i set the right version

File: src/event/data.rs (hunk 2 above)
Comment (slinkydeveloper) [reply src/event/data.rs:49]: because this method was rejecting content types like `application/json; charset=utf-8`

Comment (pranav-bhatt) [reply src/event/data.rs:49]: Ahhh, i see

**Pull request merged**